openmc/bounding_box.py

```diff
@@ -89,29 +106,84 @@ def extent(self):
     def width(self):
         return self.upper_right - self.lower_left
 
-    def extend(self, padding_distance: float) -> BoundingBox:
+    def extend(self, padding_distance: float, inplace: bool = False) -> BoundingBox:
         """Returns an extended bounding box
 
         Parameters
         ----------
         padding_distance : float
             The distance to enlarge the bounding box by
+        inplace : bool
+            Whether or not to return a new BoundingBox instance or to modify the
+            current BoundingBox object.
 
         Returns
         -------
-        An enlarged bounding box
+        An extended bounding box
+        """
+        if inplace:
+            self[0] -= padding_distance
+            self[1] += padding_distance
+            return self
+        else:
+            return BoundingBox(self[0] - padding_distance, self[1] + padding_distance)
 
+    def expand(self, other_box: BoundingBox, inplace: bool = False) -> BoundingBox:
+        """Expand the box to contain another box
+
+        Parameters
+        ----------
+        other_box : BoundingBox
+            The box used to resize this box
+        inplace : bool
+            Whether or not to return a new BoundingBox instance or to modify the
+            current BoundingBox object.
+
+        Returns
+        -------
+        An expanded bounding box
+        """
+        llc = np.minimum(self.lower_left, other_box.lower_left)
+        urc = np.maximum(self.upper_right, other_box.upper_right)
+        if inplace:
+            self.lower_left = llc
+            self.upper_right = urc
+            return self
+        else:
+            return BoundingBox(llc, urc)
+
+    def reduce(self, other_box: BoundingBox, inplace: bool = False) -> BoundingBox:
+        """Reduce the box to match the dimensions of the input bounding box
```

Comment:
Last comment (I promise!). I was going to suggest changing the name here to `intersection`, but now that I'm thinking about it we could actually implement the `__and__` and `__or__` operators instead of having the methods `reduce` and `expand`. The proper way to do this would be to also implement `__iand__` and `__ior__` and then define the `__and__` and `__or__` methods in terms of their inplace counterparts (you can look at the `Intersection` and `Union` classes to see how this is done).

Reply:
Great idea! I think that's a more elegant solution. I'll implement that shortly. Thanks for pointing me to an example elsewhere.

Reply:
All set. I thought for a bit about using `__iadd__`/`__add__` to replace the `extend` method, but I thought that the meaning of that operator might be a little ambiguous -- could be interpreted as translating/shifting the box or something like that. But if you think that would be a good change I can add that as well.

Reply:
Oh and I renamed the `extend` method to `expand` now that the name was freed up. I think it more clearly describes that operation.